plugins/sigma/src/plugin.cpp

```diff
@@ -8,17 +8,160 @@
 
 #include "sigma/parse.hpp"
 
+#include <vast/arrow_table_slice.hpp>
+#include <vast/bitmap.hpp>
+#include <vast/concept/convertible/data.hpp>
+#include <vast/concept/convertible/to.hpp>
+#include <vast/concept/parseable/string/char_class.hpp>
+#include <vast/concept/parseable/to.hpp>
+#include <vast/concept/parseable/vast/pipeline.hpp>
 #include <vast/data.hpp>
 #include <vast/error.hpp>
+#include <vast/io/read.hpp>
+#include <vast/pipeline.hpp>
 #include <vast/plugin.hpp>
+#include <vast/table_slice_builder.hpp>
 
+#include <arrow/record_batch.h>
 #include <caf/error.hpp>
 #include <caf/expected.hpp>
+#include <caf/typed_event_based_actor.hpp>
 #include <fmt/format.h>
 
+#include <thread>
+
 namespace vast::plugins::sigma {
 
-class plugin final : public virtual language_plugin {
+class sigma_operator final : public crtp_operator<sigma_operator> {
```

Comment:
After the changes from #3223 that remove language plugins we should write the changelog entry for this PR in terms of moving Sigma from a language to an operator in our language.